identitytoolkit/gitkitclient.py

```
@@ -250,7 +266,17 @@ def VerifyGitkitToken(self, jwt):
     certs = self.rpc_helper.GetPublicCert()
     crypt.MAX_TOKEN_LIFETIME_SECS = 30 * 86400  # 30 days
     try:
-      parsed = crypt.verify_signed_jwt_with_certs(jwt, certs, self.client_id)
+      parsed = None
+      for aud in [self.project_id, self.client_id]:
+        try:
+          parsed = crypt.verify_signed_jwt_with_certs(jwt, certs, aud)
+        except crypt.AppIdentityError as e:
+          if "Wrong recipient" not in e.message:
+            raise e
+        if parsed:
+          break
+      if not parsed:
+        raise crypt.AppIdentityError("Gitkit token audience doesn't match projectId or clientId in server configuration")
       return GitkitUser.FromToken(parsed)
     except crypt.AppIdentityError:
```

You mean the outer <code>try</code> block? That means exception will be propagated outside the function rather than return None as now?